### PR TITLE
Enhance error handling for NotFound error in controller reconcile.

### LINKED ISSUE #182: Controller reconcile should ignore not found error
**Describe the bug**
Some controller reconcile not ignore not found error.

### PR DESCRIPTION
**Pull request by wxyh**
fixes #182

### CODE REVIEW

File: pkg/controller/noderesource/node_resource_controller.go [hunk 1]
```diff
@@ -58,6 +59,9 @@ func (r *NodeResourceReconciler) Reconcile(ctx context.Context, req ctrl.Request
 	// get TimeSeriesPrediction result
 	err := r.Client.Get(ctx, req.NamespacedName, tsp)
 	if err != nil {
+		if errors.IsNotFound(err) {
```
Comment (qmhu) [inline pkg/controller/noderesource/node_resource_controller.go:62]: I think we can use client.IgnoreNotFound(err) too here

**Review by qmhu — COMMENTED** [review]
Nice catch! Please rebase newest code and recommit.

File: pkg/controller/noderesource/node_resource_controller.go (hunk 1 above)
Comment (wxyh) [reply pkg/controller/noderesource/node_resource_controller.go:62]: `r.Recorder.Event(tsp, v1.EventTypeNormal, "FailedGetTimeSeriesPrediction", err.Error())
		klog.Errorf("Failed to get timeSeriesPrediction(%s/%s), %v", req.NamespacedName, req.Name, err)`
		This code create event when get object error,should I delete it?

Comment (qmhu) [reply pkg/controller/noderesource/node_resource_controller.go:62]: > `r.Recorder.Event(tsp, v1.EventTypeNormal, "FailedGetTimeSeriesPrediction", err.Error()) klog.Errorf("Failed to get timeSeriesPrediction(%s/%s), %v", req.NamespacedName, req.Name, err)` This code create event when get object error,should I delete it?

I have discuss with @kitianFresh , He will refector this controller these days, so I think you can just rebase and keep keep your changes.

Comment (wxyh) [reply pkg/controller/noderesource/node_resource_controller.go]: fixed

**Pull request merged**